api/fixtures/example_kubevirt.go

```diff
@@ -0,0 +1,113 @@
+package fixtures
+
+import (
+	"fmt"
+
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/cmd/util"
+	corev1 "k8s.io/api/core/v1"
+	apiresource "k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubevirtv1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
+	capikubevirt "sigs.k8s.io/cluster-api-provider-kubevirt/api/v1alpha1"
+)
+
+func ExampleKubeVirtTemplate(o *ExampleKubevirtOptions) *hyperv1.KubevirtNodePoolPlatform {
+	runAlways := kubevirtv1.RunStrategyAlways
+	guestQuantity := apiresource.MustParse(o.Memory)
+	exampleTemplate := &hyperv1.KubevirtNodePoolPlatform{
+		NodeTemplate: &capikubevirt.VirtualMachineTemplateSpec{
+			Spec: kubevirtv1.VirtualMachineSpec{
+				RunStrategy: &runAlways,
+				Template: &kubevirtv1.VirtualMachineInstanceTemplateSpec{
+					Spec: kubevirtv1.VirtualMachineInstanceSpec{
+						Domain: kubevirtv1.DomainSpec{
+							CPU:    &kubevirtv1.CPU{Cores: o.Cores},
+							Memory: &kubevirtv1.Memory{Guest: &guestQuantity},
+							Devices: kubevirtv1.Devices{
+								Interfaces: []kubevirtv1.Interface{
+									{
+										Name: "default",
+										InterfaceBindingMethod: kubevirtv1.InterfaceBindingMethod{
+											Bridge: &kubevirtv1.InterfaceBridge{},
+										},
+									},
+								},
+							},
+						},
+						Networks: []kubevirtv1.Network{
+							{
+								Name: "default",
+								NetworkSource: kubevirtv1.NetworkSource{
+									Pod: &kubevirtv1.PodNetwork{},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	exampleTemplate.NodeTemplate.Spec.Template.Spec.Domain.Devices.Disks = []kubevirtv1.Disk{
+		{
+			Name: util.RHCOSMagicVolumeName,
+			DiskDevice: kubevirtv1.DiskDevice{
+				Disk: &kubevirtv1.DiskTarget{
+					Bus: "virtio",
+				},
+			},
+		},
+	}
+	if o.Image != "" {
+		exampleTemplate.NodeTemplate.Spec.Template.Spec.Volumes = []kubevirtv1.Volume{
+			{
+				Name: util.RHCOSMagicVolumeName,
+				VolumeSource: kubevirtv1.VolumeSource{
+					ContainerDisk: &kubevirtv1.ContainerDiskSource{
+						Image: o.Image,
+					},
+				},
+			},
+		}
+	} else {
+		dataVolume := defaultDataVolume()
+		setDataVolumeDefaults(&dataVolume, o)
+		exampleTemplate.NodeTemplate.Spec.DataVolumeTemplates = []kubevirtv1.DataVolumeTemplateSpec{dataVolume}
+	}
+	return exampleTemplate
```

Comment:
you need to set an entry in the `exampleTemplate.NodeTemplate.Spec.Template.Spec.Volumes` for this data volume

We need three things
- DataVolumeTemplate
- Volume
- Disk

right now this just lacks the volume part. I discovered this while testing.

Reply:
checking

Reply:
Fixed.